experimental/segmenter/README.md

```diff
@@ -55,46 +56,64 @@ let breakpoints: Vec<usize> = segmenter.segment_latin1(b"Hello World").collect()
 assert_eq!(&breakpoints, &[6, 11]);
 ```
 
+### Grapheme Cluster Break
+
+Segment a string:
+
+```rust
+use icu_segmenter::GraphemeClusterBreakIterator;
+
+let breakpoints: Vec<usize> = GraphemeClusterBreakIterator::new("Hello World").collect();
+assert_eq!(&breakpoints, &[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
+```
+
+Segment a Latin1 byte string:
+
+```rust
+use icu_segmenter::GraphemeClusterBreakIteratorLatin1;
+
+let breakpoints: Vec<usize> = GraphemeClusterBreakIteratorLatin1::new(b"Hello World").collect();
+assert_eq!(&breakpoints, &[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
+```
+
 ### Word Break
 
-Segment a string with default options:
+Segment a string:
 
 ```rust
 use icu_segmenter::WordBreakIterator;
 
-let mut iter = WordBreakIterator::new("Hello World");
-let result: Vec<usize> = iter.collect();
-println!("{:?}", result);
+let breakpoints: Vec<usize> = WordBreakIterator::new("Hello World").collect();
+assert_eq!(&breakpoints, &[0, 5, 6, 11]);
```

Comment:
`question`: I noticed that all segmenter APIs return the start and end of the string as a break opportunity. What is the rationale behind it? Is there a scenario where the start/end of a string is not a viable break opportunity?
If not, would it be hard to remove unnecessary result values?

Reply:
heh, I asked @makotokato the same question in https://github.com/unicode-org/icu4x/pull/1500.

According to the spec, [SB1](https://www.unicode.org/reports/tr29/#Sentence_Boundary_Rules), [WB1](https://www.unicode.org/reports/tr29/#Word_Boundary_Rules), and [GB1](https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundary_Rules) in UAX29 all say "Break at the start and end of text, unless the text is empty.", so it seems reasonable to return both start and end of the string as a break opportunity. However, notice that for line breaking, [LB2](https://www.unicode.org/reports/tr14/#Algorithm) is "Never break at the start of text.".

ICU4C's UAX29 breakers return the end line of string as a break opportunity, but not the start of the string, which seems deviate from the spec. I'm not sure whether we want to align the behavior or not. Personally, I feel including break opportunity at end of string is useful, because we can the assemble a substring using [currentBreakPos, nextBreakPos).

Reply:
@markusicu - is there a reason for the spec to mandate start/end?